lms/djangoapps/grades/scores.py

```diff
@@ -5,106 +5,242 @@
+    """
+    if persisted_block:
+        return persisted_block.graded
+    else:
+        return _get_explicit_graded(block)
 
-    if not user.is_authenticated():
-        return (None, None)
 
-    location_url = unicode(block.location)
-    if location_url in submissions_scores_cache:
-        return submissions_scores_cache[location_url]
+def _get_explicit_graded(block):
+    """
+    Returns the explicit graded field value for the given block.
+    """
+    field_value = getattr(
+        block.transformer_data[GradesTransformer],
+        GradesTransformer.EXPLICIT_GRADED_FIELD_NAME,
+        None,
+    )
 
-    if not getattr(block, 'has_score', False):
-        # These are not problems, and do not have a score
-        return (None, None)
+    # Set to True if grading is not explicitly disabled for
+    # this block.  This allows us to include the block's score
+    # in the aggregated self.graded_total, regardless of the
+    # inherited graded value from the subsection. (TNL-5560)
+    return True if field_value is None else field_value
 
-    # Check the score that comes from the ScoresClient (out of CSM).
-    # If an entry exists and has a total associated with it, we trust that
-    # value. This is important for cases where a student might have seen an
-    # older version of the problem -- they're still graded on what was possible
-    # when they tried the problem, not what it's worth now.
-    score = scores_client.get(block.location)
-    if score and score.total is not None:
-        # We have a valid score, just use it.
-        earned = score.correct if score.correct is not None else 0.0
-        if possible is None:
-            possible = score.total
-        elif possible != score.total:
-            log.error(
-                u"Persistent Grades: scores.get_score, possible value {} != score.total value {}".format(
-                    possible,
-                    score.total
-                )
-            )
+
+def _weighted_score(r_earned, r_possible, weight):
+    """
+    Returns a tuple that represents the weighted (earned, possible) score.
+    If weight is None or r_possible is 0, returns the original values.
+    """
+    assert r_possible is not None
+    cannot_compute_with_weight = weight is None or r_possible == 0
+    if cannot_compute_with_weight:
+        return r_earned, r_possible
     else:
-        # This means we don't have a valid score entry and we don't have a
-        # cached_max_score on hand. We know they've earned 0.0 points on this.
-        earned = 0.0
-        if possible is None:
-            possible = block.transformer_data[GradesTransformer].max_score
-
-        # Problem may be an error module (if something in the problem builder failed)
-        # In which case possible might be None
-        if possible is None:
-            return (None, None)
-
-    return weighted_score(earned, possible, weight)
+        return float(r_earned) * weight / r_possible, float(weight)
```

Comment:
I'm a little confused here, the documentation and `cannot_compute_with_weight` return led me to expect a tuple of weighted `(earned, possible)` values, but it looks like we're returning a tuple of `(weighted percentage, weight)`?